src/Twilio/Security/RequestValidator.php

```diff
@@ -67,35 +67,41 @@ public static function computeBodyHash($data = '') {
      * @param string $expectedSignature
      * @param string $url
      * @param array $data
-     * @return bool 
+     * @return bool
      */
 
     public function validate($expectedSignature, $url, $data = array()) {
 
-        if (\is_array($data)) {
+        $urlWithPort = RequestValidator::addPort(\parse_url($url));
+        $urlWithoutPort = RequestValidator::removePort(\parse_url($url));
 
-            return self::compare(
-                $this->computeSignature($url, $data),
-                $expectedSignature
-            );
-
-        } else {
+        $validBodyHash = true;  // May not receive body hash, so default succeed
 
+        if (!\is_array($data)) {
             // handling if the data was passed through as a string instead of an array of params
             $queryString = \explode('?', $url);
             $queryString = $queryString[1];
             \parse_str($queryString, $params);
 
-            return self::compare(
-                $this->computeSignature($url),
-                $expectedSignature
-            ) && self::compare(
-                self::computeBodyHash($data),
-                $params['bodySHA256']
-            );
-
+            $params['bodySHA256'] = (isset($params['bodySHA256'])) ? $params['bodySHA256'] : NULL;
```

Comment:
Why is this line needed?

Reply:
Was getting an "risky test" error when running `make test` since `params['bodySHA256']` is not always defined. PHP manual states that accessing an undefined key will raise an E-NOTICE level error message and return `NULL`. This line ensures the `NULL` return value while avoiding the error, since this behavior is common.

Reply:
Just found this util: https://github.com/twilio/twilio-php/blob/master/src/Twilio/Values.php#L12

Reply:
oh good, I'll refactor to include that.